cmd/charm/charmcmd/cmd.go

```diff
@@ -523,3 +548,15 @@ func (d *progressDisplay) stopMonitor() {
 		d.printer.Done()
 	}
 }
+
+func readAgentFile(f string) (*agent.AuthInfo, error) {
```

Comment:
Isn't there a function in the bakery to do this? there probably should be.

Reply:
Yeah, there probably should. I can't quite work out whether it might be best combined into one function. For now, I think I'll leave it like this and update it later.